decimal.go

```diff
@@ -95,7 +106,7 @@ func NewFromString(value string) (Decimal, error) {
 			// NOTE(vadim): I doubt a string could realistically be this long
 			return Decimal{}, fmt.Errorf("can't convert %s to decimal: fractional part too long", value)
 		}
-		exp = int32(expInt)
+		exp += int32(expInt)
```

Comment:
Since you're adding `expInt` to `exp` here (both int32s), there could potentially be an int32 overflow. I think a good way to handle this would be to make `exp` an int64, and then right before you create and return the Decimal, to return an error if it is < math.MinInt32 or > math.MaxInt32.

You can then remove the the `if expInt < math.MinInt32` above this line.